### PR TITLE
Feat/table basic

### PR DESCRIPTION
**Pull request by nucliweb**
# In this Pull Request

New `<TableBasic />` component.

preview

## API

```
TableBasic.propTypes = {
  contentHead: PropTypes.array,
  contentBody: PropTypes.arrayOf(
    PropTypes.arrayOf(
      PropTypes.shape({
        content: PropTypes.string.isRequired,
        type: PropTypes.oneOf(Object.values(CELL_TYPE)),
        isNowrap: PropTypes.bool
      })
    )
  ).isRequired,
  contentFoot: PropTypes.array
}
```

### CODE REVIEW

File: components/table/basic/src/index.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+
+const CELL_TYPE = {
+  header: 'th',
+  data: 'td'
+}
+
+const TableBasic = ({contentHead, contentBody, contentFoot}) => {
+  const hasHead = Boolean(contentHead?.length)
```
Comment (jordevo) [inline components/table/basic/src/index.js]: Optional Chaining FTW 🎉

File: components/table/basic/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+```sh
+$ npm install @s-ui/sui-table-basic --save
+```
+
+## Usage
+
+### Basic usage
+
+```js
+import TableBasic from '@s-ui/sui-table-basic'
```
Comment (midudev) [inline components/table/basic/README.md]: Naming is not correct. It should be: `@s-ui/react-atom-basic`.

File: components/table/basic/package.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "@s-ui/react-table-basic",
```
Comment (midudev) [inline components/table/basic/package.json]: You've it right here! 👍

File: components/table/basic/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+return <TableBasic />
```
Comment (midudev) [inline components/table/basic/README.md]: Maybe a simple demo could help!

File: components/table/basic/src/index.scss [hunk 5]
```diff
@@ -0,0 +1,24 @@
+@import '~@schibstedspain/sui-theme/lib/index';
+
+.sui-TableBasic {
+  border-collapse: collapse;
+  font-size: $fz-s;
+  padding: 0;
+  color: $c-gray-dark;
+
+  &-header {
+    th {
```
Comment (midudev) [inline components/table/basic/src/index.scss]: You could use the className here instead.

File: components/table/basic/src/index.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  const hasFoot = Boolean(contentFoot?.length)
+
+  return (
+    <table className="sui-TableBasic sui-TableBasic--noWrap">
```
Comment (davecarter) [inline components/table/basic/src/index.js]: you can declare a `baseClass` and then reuse it everywhere combined with template strings as follows:

```javascript
const baseClass = 'sui-TableBasic'

return (
    <table className={`${baseClass}  ${baseClass}--noWrap`}>
    (...)
```

Comment (midudev) [inline components/table/basic/src/index.js]: The `noWrap` seems to be not used on the CSS. I think is safe to remove.

File: components/table/basic/src/index.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+                {element}
+              </th>
+            ))}
+          </tr>
+        </thead>
+      )}
+
+      <tbody>
+        {contentBody.map((row, index) => {
+          return (
```
Comment (AgonisticKatai) [inline components/table/basic/src/index.js]: not needed with arrow function
```javascript
{contentBody.map((row, index) => (
    <tr key={index}>
    [...]
)}

File: components/table/basic/src/index.js [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+              )
+            })}
+          </tr>
+        ))}
+      </tbody>
+
+      {hasFoot && (
+        <tfoot className={`${baseClass}-header`}>
+          <tr>
+            {contentHead.map((element, index) => (
```
Comment (JoanClaret) [inline components/table/basic/src/index.js]: should that be `contentFoot` ?

Comment (nucliweb) [reply components/table/basic/src/index.js]: Yes, well see, thanks

File: components/table/basic/src/index.js [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+        </tfoot>
+      )}
+    </table>
+  )
+}
+
+TableBasic.displayName = 'TableBasic'
+
+TableBasic.propTypes = {
+  contentHead: PropTypes.array,
```
Comment (kikoruiz) [inline components/table/basic/src/index.js]: I would expect better `head`, `body` and `footer` as props, don't you think?

File: components/table/basic/src/index.js [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+                  {cell.content}
+                </Element>
+              )
+            })}
+          </tr>
+        ))}
+      </tbody>
+
+      {hasFoot && (
+        <tfoot className={`${baseClass}-header`}>
```
Comment (kikoruiz) [inline components/table/basic/src/index.js]: Shouldn't be `-foot` here?

**Comment by ivanmlaborda** [comment]
Good job!! :clap:

**Pull request merged**